### PR TITLE
Cin delta 155 view all projects

### PR DESCRIPTION
**Pull request by tonymagne**

### CODE REVIEW

File: src/main/scala/com/focaldata/cinprojectsurvey/graphql/surveys/SurveysGraphQL.scala [hunk 1]
```diff
@@ -39,7 +39,7 @@ object SurveysGraphQL {
         env <- ZIO.environment[Env]
         id <- env.surveys.generateId
         now <- env.surveys.now
-        nickname <- env.surveys.generateNickname(surveyCreateInput.name, id)
+        nickname = Surveys.makeNickname(surveyCreateInput.name, id)
```
Comment (peterlogg) [inline src/main/scala/com/focaldata/cinprojectsurvey/graphql/surveys/SurveysGraphQL.scala]: @tonymagne could you explain to us what is going on here please, why `def makeNickname` is on a different footing to the other methods within the survey repo.

File: src/main/scala/com/focaldata/cinprojectsurvey/repositories/projects/Projects.scala [hunk 2]
```diff
@@ -1,34 +1,192 @@
...
+      def getAllSolutions: ZIO[Env,Nothing,List[Solution]] = for {
+        env       <- ZIO.environment[Doobie]
+        solutions <- allSolutionsQuery.to[List].transact(env.xa).orDie
+      } yield solutions
+
+      def createProject(projectId: ProjectId, input: ProjectCreateInput, surveyId: SurveyId, defaultSurvey: DefaultDraftSurvey, sinceDate: Instant): ZIO[Env, ProjectAlreadyExists, Project] = for {
+        env           <- ZIO.environment[Doobie]
+        nickname      =  makeNickname(input.name, projectId)
+        (projectTuple, solution) <- insertProjectAndDraftSurvey(projectId, nickname, input, surveyId, defaultSurvey, sinceDate).transact(env.xa).attemptSomeSqlState { 
+                                      case sqlstate.class23.UNIQUE_VIOLATION => ProjectAlreadyExists(projectId)
```
Comment (peterlogg) [inline src/main/scala/com/focaldata/cinprojectsurvey/repositories/projects/Projects.scala]: @tonymagne could you explain what is being done here, please

File: src/main/scala/com/focaldata/cinprojectsurvey/graphql/surveys/SurveysGraphQL.scala (hunk 1 above)
Comment (tonymagne) [reply src/main/scala/com/focaldata/cinprojectsurvey/graphql/surveys/SurveysGraphQL.scala]: Yes. It's a pure function (no effects) so there's no need to have it returning a ZIO.
All it does is to take name and id and return a nickname.

File: src/main/scala/com/focaldata/cinprojectsurvey/repositories/projects/Projects.scala (hunk 2 above)
Comment (tonymagne) [reply src/main/scala/com/focaldata/cinprojectsurvey/repositories/projects/Projects.scala]: That's a doobie feature to catch constraints violation from the db.
Unfortunately you don't get the actual constraint (each constraint has name in our schema).
That's a doobie limitation (probably a JDBC limitation)

File: src/test/scala/com/focaldata/cinprojectsurvey/repositories/surveys/SurveysSpec.scala [hunk 3]
```diff
@@ -1,48 +1,49 @@
...
 
-  def mkSurveyTestEnv =
-    for {
-      config     <- postgresqlConfig.load[Task].toManaged_
-      _          <- migratePostgres(config).toManaged_
-      transactor <- mkManagedPostgresqlHikariTransactor(config)(blockingEC)
-    } yield new Doobie {
-      override val xa = transactor
-    }
+  def makeSurveyTestWithcContainerEnv = for {
```
Comment (peterlogg) [inline src/test/scala/com/focaldata/cinprojectsurvey/repositories/surveys/SurveysSpec.scala]: typo : ...WithcContainerEnv -> ...cContainer...

File: src/main/scala/com/focaldata/cinprojectsurvey/graphql/surveys/SurveysGraphQL.scala (hunk 1 above)
Comment (peterlogg) [reply src/main/scala/com/focaldata/cinprojectsurvey/graphql/surveys/SurveysGraphQL.scala]: OK, I see what you mean, thanks.

File: src/main/scala/com/focaldata/cinprojectsurvey/repositories/projects/Projects.scala (hunk 2 above)
Comment (peterlogg) [reply src/main/scala/com/focaldata/cinprojectsurvey/repositories/projects/Projects.scala]: great thanks

**Pull request merged**